### PR TITLE
Code review and and tests adjustments

### PR DESCRIPTION
**Pull request by NicholasTurner23**
Just some cleanup - For PR-6522

## Summary by CodeRabbit

* **New Features**
  * Implemented batched database operations for improved persistence performance.
  * Enhanced hourly site-level forecasting with improved metadata aggregation.

* **Bug Fixes**
  * Tightened error validation and messaging for missing or insufficient input data.
  * Added soft-failure handling for meteorology data enrichment.

* **Tests**
  * Expanded test coverage for hourly forecast generation, validation, and edge cases.
  * Added tests for meteorology data integration scenarios.

* **Refactor**
  * Reorganized forecast job orchestration across modules.

### CODE REVIEW

File: src/workflows/dags/AirQo-site-HOURLY-forecasting-job_Q.py
Comment (NicholasTurner23) [inline src/workflows/dags/AirQo-site-HOURLY-forecasting-job_Q.py:1]: Moved to forecast_prediction_jobs.py @wabinyai

File: src/workflows/airqo_etl_utils/ml_utils.py [hunk 1]
```diff
@@ -2406,7 +2409,7 @@ def generate_site_hourly_forecasts(
             recursive_history
         )
         predictions: List[pd.DataFrame] = []
-
+        # Review operation for optimization. Depending on the number of sites and the forecast horizon, this loop could be a bottleneck. Potential optimizations include vectorizing the candidate generation and prediction steps to handle all sites in bulk rather than iterating through each hour sequentially.
```
Comment (NicholasTurner23) [inline src/workflows/airqo_etl_utils/ml_utils.py:2412]: @wabinyai take note of this comment. Loops inside of loops are discouraged as they can worsen complexity(time and space).
